runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -11790,53 +11788,130 @@ static bool inlineIsAssignableFrom(TR::Node *node, TR::CodeGenerator *cg)
    return true;
    }
 
+static TR::SymbolReference *getClassSymRefAndDepth(TR::Node *classNode, TR::Compilation *comp, int32_t &classDepth)
+   {
+   classDepth = -1;
+    // recognizedCallTransformer adds another layer of aloadi
+   // need to loop find non-aloadi child
+   while (classNode->getOpCodeValue() == TR::aloadi && classNode->getNumChildren() > 0 && classNode->getFirstChild()->getOpCodeValue() == TR::aloadi)
+      {
+      classNode = classNode->getFirstChild();
+      }
+
+   if ((classNode->getOpCodeValue() != TR::aloadi) || (classNode->getSymbolReference() != comp->getSymRefTab()->findJavaLangClassFromClassSymbolRef()) ||
+         (classNode->getFirstChild()->getOpCodeValue() != TR::loadaddr)) return NULL;
+
+   TR::SymbolReference *classSymRef = NULL;
+   TR::Node *classRef = classNode->getFirstChild();
```

Comment:
Are we guaranteed to classNode with a child ? We are dereferencing classRef later in this function, if it is possible to be NULL, then we could end up with some seg - faults.

If you think we can never have NULL, please add a FATAL assert to ensure we do not miss any case.